testing/src/main/java/net/consensys/linea/testing/GeneralStateReferenceTestTools.java

```diff
@@ -126,6 +126,7 @@ public static void executeTest(
 
       transactionProcessingResultValidator.accept(transaction, result);
       zkTracerValidator.accept(tracer);
+      worldStateUpdater.commit();
```

Comment:
This is I guess what allows you to extract the right `nonce` in the next transaction ?

Reply:
Now I'm confused: we do the commit at the end of `executeTest`, e.g. for multi block tests, after all blocks were processed. Why do that instead of after every block as was previously the case ?